src/Maker/MakeAuthenticator.php

```diff
@@ -55,50 +70,94 @@ public function configureCommand(Command $command, InputConfiguration $inputConf
     {
         $command
             ->setDescription('Creates an empty Guard authenticator')
-            ->addArgument('authenticator-class', InputArgument::OPTIONAL, 'The class name of the authenticator to create (e.g. <fg=yellow>AppCustomAuthenticator</>)')
             ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeAuth.txt'));
     }
 
     public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
     {
-        if (!$this->fileManager->fileExists($path = 'config/packages/security.yaml')) {
-            return;
-        }
+        // authenticator type
+        $authenticatorTypeValues = [
+            'Empty authenticator' => self::AUTH_TYPE_EMPTY_AUTHENTICATOR,
+            'Form login' => self::AUTH_TYPE_FORM_LOGIN,
+        ];
+        $command->addArgument('authenticator-type', InputArgument::REQUIRED);
+        $authenticatorType = $io->choice(
+            'Which authentication type do you want ?',
+            array_keys($authenticatorTypeValues),
+            key($authenticatorTypeValues)
+        );
+        $input->setArgument(
+            'authenticator-type',
+            $authenticatorTypeValues[$authenticatorType]
+        );
 
-        $manipulator = new YamlSourceManipulator($this->fileManager->getFileContents($path));
+        $manipulator = new YamlSourceManipulator($this->fileManager->getFileContents('config/packages/security.yaml'));
         $securityData = $manipulator->getData();
 
-        $interactiveSecurityHelper = new InteractiveSecurityHelper();
-
-        $command->addOption('firewall-name', null, InputOption::VALUE_OPTIONAL, '');
-        $input->setOption('firewall-name', $firewallName = $interactiveSecurityHelper->guessFirewallName($io, $securityData));
-
-        $command->addOption('entry-point', null, InputOption::VALUE_OPTIONAL);
+        if (self::AUTH_TYPE_FORM_LOGIN === $input->getArgument('authenticator-type')
+            && !isset($securityData['security']['providers']) || !$securityData['security']['providers']) {
+            throw new RuntimeCommandException('You need to have at least one provider defined in security.yaml');
```

Comment:
This is pretty similar than the next comment you've made : once the user has chosen "form login", we already know that we should raise an exception, but, if i put it down before `InteractiveSecurityHelper::guessUserClass()`, we're asking 4 questions before raising the exception... i think this could be annoying for the user.
What's your feeling about that ? - but i'm ok that in the code it seems unrelated to anything... at the beginning this test was right before `guessUserClass()` call...

ok for other comments